util/generate-rsa-signing-tests.py

```diff
@@ -40,57 +51,161 @@ def parse(fn):
                 continue
             name, val = ln.split('=', 1)
             cur[name.strip()] = val.strip()
-            if name.strip() == 'S':
+            if name.strip() == last_field:
                 cases.append(cur)
                 cur = copy.copy(cur)
     return cases
 
-def main(fn):
-    for case in parse(fn):
+def main(fn, test_type, padding_alg):
+    # File header
+    print "# NIST RSA PSS Test Vectors for FIPS 186-4 from %s in\n" % fn + \
+    "# http://csrc.nist.gov/groups/STM/cavp/documents/dss/186-3rsatestvectors.zip\n" + \
+    "# accessible from" + \
+    "# http://csrc.nist.gov/groups/STM/cavp/digital-signatures.html#test-vectors.\n" + \
+    "# Generated with the help of the util/generate-rsa-signing-tests.py script.\n" + \
+    "#\n" + \
+    "# Digest = SHAAlg.\n" + \
+    "# Key is a DER-formatted PKCS1-encoded RSA key.\n" + \
+    "# Sig = S.\n\n"
+
+    num_cases = 0
+
+    # Each test type has a different field as the last entry per case
```

Comment:
This if-block seems a bit iffy, since it uses if/elif/else for conditions that don't seem mutually exclusive. Maybe add a comment on each condition line to clarify? Also, I would add an empty line after the if-block, and initialize `num_cases` after it (right before the for-loop).